File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function solution(balls, share) {
    return Math.floor(fac(share+1, balls) / (fac(2, balls-share)*fac(share)));
}

function fac(s, e) {
    var value=1;
    for(var i=s; i<=e; i++){
        value*=i;
    }
    return value;
}

File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
class Solution {
    public int solution(int balls, int share) {
        double answer = 1;
        for(int i=balls-share; i>=1; i--,balls--){
            answer *= balls;
            System.out.println("1. "+answer);
            answer /= i;
            System.out.println("2. "+answer);
        }
        return (int)Math.round(answer);
    }
}